src/include/mempool.h

```diff
@@ -376,7 +376,7 @@ class pool_allocator {
                                                                         \
     template<typename k,typename v, typename cmp = std::less<k> >	\
     using map = std::map<k, v, cmp,					\
-			 pool_allocator<std::pair<k,v>>>;		\
+			 pool_allocator<std::pair<const k,v>>>;		\
```

Comment:
@tchaikov 
I held back on that because I wasn't sure the issue is there as well.
But I'll put them in as well with these routines.